com.microsoft.mrtk.input/InteractionModes/FlatScreenModeDetector.cs

```diff
@@ -0,0 +1,53 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.XR.Interaction.Toolkit;
+
+namespace Microsoft.MixedReality.Toolkit.Input
+{
+    internal class FlatScreenModeDetector : MonoBehaviour, IInteractionModeDetector
+    {
+        [SerializeField]
+        [Tooltip("The interaction mode to be set when the interactor is hovering over an interactable.")]
+        private InteractionMode flatScreenInteractionMode;
+
+        [SerializeField]
+        [Tooltip("List of XR Base Controllers that this interaction mode detector has jurisdiction over. Interaction modes will be set on all specified controllers.")]
+        private List<GameObject> controllers;
+
+        public InteractionMode ModeOnDetection => flatScreenInteractionMode;
+
+        protected ControllerLookup controllerLookup = null;
+
+        public void Awake()
+        {
+            ControllerLookup[] lookups = GameObject.FindObjectsOfType(typeof(ControllerLookup)) as ControllerLookup[];
+            if (lookups.Length == 0)
+            {
+                Debug.LogError("Could not locate an instance of the ControllerLookup class in the hierarchy. It is recommended to add ControllerLookup to your camera rig.");
```

Comment:
Will a lack of ControllerLookup break the rig? If so, this should probably be a bit more stringent that "it is recommended", or maybe even add a ControllerLookup automatically?

Reply:
ControllerLookup is already baked into the rig but I can make it an automatic add in conjuction with the InteractionModeManager for now. I think we need a better solution for controller lookup in the long term.

Reply:
To be clear, I think a separate PR would be appropriate for tackling the controller lookup question